plugins/orchestrator/src/api/OrchestratorClient.ts

```diff
@@ -78,11 +104,24 @@ export class OrchestratorClient implements OrchestratorApi {
     );
   }
 
-  async listWorkflowOverviews(): Promise<WorkflowOverviewListResult> {
-    const baseUrl = await this.getBaseUrl();
-    return await this.fetcher(`${baseUrl}/workflows/overview`).then(r =>
-      r.json(),
-    );
+  async listWorkflowOverviews(
+    paginationInfo?: PaginationInfoDTO,
+    testDefaultApi?: DefaultApi, // to make the function testable
+  ): Promise<AxiosResponse<WorkflowOverviewListResultDTO>> {
+    const defaultApi = testDefaultApi || (await this.getDefaultAPI());
+    const reqConfigOption: AxiosRequestConfig =
+      await this.getDefaultReqConfig();
+    try {
+      return await defaultApi.getWorkflowsOverview(
+        paginationInfo?.page,
+        paginationInfo?.pageSize,
+        paginationInfo?.orderBy,
+        paginationInfo?.orderDirection,
+        reqConfigOption,
+      );
+    } catch (error: any) {
+      throw new Error(error);
```

Comment:
no point in just catching an error and throwing it again as is...

Reply:
addressed